src/Interpreters/executeQuery.cpp

```diff
@@ -687,7 +743,8 @@ void logQueryFinish(
     bool internal)
 {
     const auto time_now = std::chrono::system_clock::now();
-    logQueryFinishImpl(elem, context, query_ast, std::move(query_pipeline), pulling_pipeline, query_span, query_result_cache_usage, internal, time_now);
+    auto query_pipeline_finalized_info = finalizeQueryPipelineBeforeLogging(std::move(query_pipeline), query_result_cache_usage, pulling_pipeline);
```

Comment:
Open question: Should `time_now` come after and include the time spent in `finalizeQueryPipelineBeforeLogging`?